### PR TITLE
Auth - openid authorization and token refresh

### PR DESCRIPTION
**Pull request by davidwincent**

### CODE REVIEW

**Comment by johnnyblasta** [comment]
Interesting, but please tell us more and make a issue.

**Comment by davidwincent** [comment]
Issue #84 

We are also working on a plugin for OpenID authorization with Origo.

The server part is needed to support authorization with 3d part vendors and code authorization flow / token refresh.

**Comment by asemoller** [comment]
Settings in conf/config.js:
```
auth: {
    openidIssuer: 'https://....', (url to openID connect provider)
    redirect_uri: 'http://xxx/origoserver/auth', (url to origo server endpoint for authorization)
    client_id: 'xxx',
    client_secret: 'xxx'
  }
```

Can be used with the [plugin for OpenID connect](https://github.com/SigtunaGIS/oidc-plugin) for Origo. Note that an openID connect provider is required for this to work.

**Comment by Grammostola** [comment]
We shall have reason to test this within "the next month or so."

File: handlers/auth/access_token.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      res.status(400).send('Bad Request: Neither code nor refresh token found.');
+    }
+    const user_info = await client.userinfo(token_set.access_token);
+    res.json({
+      authenticated: true,
+      access_token: token_set.access_token,
+      refresh_token: token_set.refresh_token,
+      id_token: token_set.id_token,
+      expires_at: token_set.expires_at,
+      displayname: user_info[conf.auth.displayname_property]
```
Comment (tablackmore) [inline handlers/auth/access_token.js:33]: user_info[conf.auth.displayname_property]

displayname_property doesn't exist in the config

Shouldn't it be:

displayname: user_info[conf.auth.display_name]

**Review by tablackmore — CHANGES_REQUESTED** [review]

**Comment by Grammostola** [comment]
We also saw that with our ADFS idp 
`user_info[conf.auth.display_name` doesn't work (either) in that no username is populated in the Origo   / oidc-plugin UI, that is when the auth config is set as `display_name: 'unique_name',`
however changing access_token.js that is referenced above like so:
`displayname: token_set.claims().unique_name` seems to work - the unique_name is populated in the UI.

**Comment by jokd** [comment]
displayname_property: 'email' works with google service at least

File: handlers/auth/access_token.js (hunk 1 above)
Comment (jokd) [reply handlers/auth/access_token.js:33]: Agree on this^^

**Comment by jokd** [comment]
Moved to #118 
Thanks @davidwincent

**Pull request closed without merging**